source/declarations.tex

```diff
@@ -227,12 +227,12 @@
 
 \pnum
 A \grammarterm{nodeclspec-function-declaration} shall declare a
-constructor, destructor, or conversion function.  \footnote{The
+constructor, destructor, or conversion function.\footnote{The
 ``implicit int'' rule of C is no longer supported.}
 \begin{note}
 A \grammarterm{nodeclspec-function-declaration} can only be used in a
 \grammarterm{template-declaration}~(Clause~\ref{temp}),
-\grammarterm{explicit-instantiation}~(\ref{temp.explicit}), or
+\grammarterm{explicit-instan\-tiation}~(\ref{temp.explicit}), or
```

Comment:
Right, but as I said that's rather strange if it requires hyphenation in the previous term. I think the non-breaking space is only an approximation, and what we really want is a "discouraging" space that says "please try not to break here if possible".

But surely it's *much* better to break at a real space than introduce awkward hyphenation. What do you think? In real-world publishing you would probably try to rephrase the text to achieve the desired presentation, but given the constant changes we are making, I would strongly prefer the simple and maintainable solution that doesn't require manual intervention.